java8/src/main/java/feign/optionals/OptionalDecoder.java

```diff
@@ -22,8 +22,7 @@ public OptionalDecoder(Decoder delegate) {
         if(!isOptional(type)) {
             return delegate.decode(response, type);
         }
-
-        if(response.status() == 404) {
+        if(response.status() == 404 || response.status() == 204) {
```

Comment:
Potentially even more statuses could be added, in which case it should be rewritten to something like 

```
if (Stream.of(404, 204).anyMatch(status -> status == response.status()) { ... }
```

Reply:
At the moment, we are keeping `feign` core at Java 6+.  Java 8 Specific features are available in the `feign-java8` module.